### PR TITLE
chore(RichTextEditor): add `status` prop

### PR DESCRIPTION
**Pull request by ozgurkececioglu**
[FX-2510]

### Description

- added `status` prop just like other input props

### How to test

- you can check the form page to see its working

### Screenshots

no visual changes

### Review

- [ ] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [ ] Annotate all `props` in component with documentation
- [ ] Create `examples` for component
- [ ] Ensure that deployed demo has expected results and good examples
- [ ] Ensure that tests pass by running `yarn test`
- [ ] Ensure that visuals tests pass by running `yarn test:visual`. If not - check the documentation [how to fix visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md#fixing-broken-visual-tests-inside-a-pr)
- [ ] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run all` - Run whole pipeline
- `@toptal-bot run build` - Check build
- `@toptal-bot run visual` - Run visual tests
- `@toptal-bot run deploy:documentation` - Deploy documentation
- `@toptal-bot run package:alpha-release` - Release alpha version

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-2510]: https://toptal-core.atlassian.net/browse/FX-2510?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/picasso/src/RichTextEditor/RichTextEditor.tsx [hunk 1]
```diff
@@ -107,17 +112,24 @@ export const RichTextEditor = forwardRef<HTMLDivElement, Props>(
...
     const wrapperRef = useRef<HTMLDivElement | null>(null)
     const { dispatch, state } = useTextEditorState()
 
+    usePropDeprecationWarning({
+      props,
+      name: 'error',
+      componentName: 'RichTextEditor',
+      description:
+        'Use the `status` prop instead. `error` is deprecated and will be removed in the next major release.'
+    })
```
Comment (dmaklygin) [inline packages/picasso/src/RichTextEditor/RichTextEditor.tsx:131]: 👍

File: packages/picasso/src/RichTextEditor/story/index.jsx [hunk 2]
```diff
@@ -20,3 +20,4 @@ page
   .addExample('RichTextEditor/story/DefaultValue.example.tsx', 'Default value') // picasso-skip-visuals
   .addExample('RichTextEditor/story/Disabled.example.tsx', 'Disabled') // picasso-skip-visuals
   .addExample('RichTextEditor/story/Limit.example.tsx', 'Limit Length') // picasso-skip-visuals
+  .addExample('RichTextEditor/story/Status.example.tsx', 'Status') // picasso-skip-visuals
```
Comment (dmaklygin) [inline packages/picasso/src/RichTextEditor/story/index.jsx:23]: shouldn't we somehow cover status with visual tests?

File: packages/picasso/src/RichTextEditor/RichTextEditor.tsx [hunk 3]
```diff
@@ -39,9 +41,12 @@ export interface Props extends BaseProps {
   /** unique identifier */
   id: string
   /**
-   * Indicate wether the editor is in an error state
+   * @deprecated Use the `status` prop instead to both support success and error states
+   * Indicate whether `RichTextEditor` is in error state
    */
   error?: boolean
+  /** Indicate `RichTextEditor` status */
+  status?: Exclude<Status, 'success'>
```
Comment (dmaklygin) [inline packages/picasso/src/RichTextEditor/RichTextEditor.tsx:49]: this is confusing. I would manually describe all possible values of status here. wdyt?

File: cypress/component/RichTextEditor.spec.tsx [hunk 4]
```diff
@@ -84,6 +84,14 @@ describe('RichTextEditor', () => {
     cy.get('body').happoScreenshot()
   })
 
+  describe('when in an invalid state', () => {
+    it('shows error', () => {
+      mount(renderEditor({ ...defaultProps, status: 'error' }))
+
+      cy.get('body').happoScreenshot()
+    })
+  })
```
Comment (dmaklygin) [inline cypress/component/RichTextEditor.spec.tsx:93]: 👍

**Review by denieler — CHANGES_REQUESTED** [review]
we need to limit `Status` typing or add `success` variant to the example

File: packages/picasso/src/Select/types.ts [hunk 5]
```diff
@@ -32,8 +32,8 @@ export interface SelectProps<
   disabled?: boolean
   /** @deprecated Indicate whether `Select` is in error state */
   error?: boolean
-  /** Indicate whether `Select` is in error state */
-  status?: Exclude<Status, 'success'>
+  /** Indicate whether `Select` is in `error` or `default` state */
+  status?: Status
```
Comment (denieler) [inline packages/picasso/src/Select/types.ts:36]: 🤔

Comment (denieler) [reply packages/picasso/src/Select/types.ts:36]: to me actually `'error' | 'default'` look good 🤔

Comment (denieler) [reply packages/picasso/src/Select/types.ts:36]: what about `Pick<Status, 'error' | 'default'>`?

Comment (ozgurkececioglu) [reply packages/picasso/src/Select/types.ts:36]: Slightly safer 😄 I would be ok with that. WDYT @dmaklygin ?

Comment (ozgurkececioglu) [reply packages/picasso/src/Select/types.ts:36]: @denieler new info for today. `Pick` won't work with literal types 🙂 I surprised also:
[docs](https://www.typescriptlang.org/docs/handbook/utility-types.html#picktype-keys)
[playground example](https://www.typescriptlang.org/play?#code/C4TwDgpgBAyglgLwgFXNAvFA5AWwgEzgFccsoAfbAZxwEMAbesyre2gJwHMIsAoXuADtgEdgDNaAY2gAFdgHswVKAG9eUDVCqIIAfgBcUGXEkBrADzwkqSABpseQiWbU6jLAD5eAXyA)

Comment (denieler) [reply packages/picasso/src/Select/types.ts]: try `Extract` then

**Review by TomasSlama — APPROVED** [review]
GJ 👍

**Review by denieler — APPROVED** [review]

**Review by LashaJini — APPROVED** [review]
Tested locally and works fine :+1: .

File: packages/picasso/src/Select/types.ts (hunk 5 above)
Comment (dmaklygin) [reply packages/picasso/src/Select/types.ts]: yeah, Exctract is more reasonable here :)

**Pull request merged**